flow/project.py

```diff
@@ -1601,13 +1870,31 @@ def _generate_operations(self, cmd, jobs, requires=None):
             cmd_ = cmd.format(job=job)
             yield JobOperation(name=cmd_.replace(' ', '-'), cmd=cmd_, job=job)
 
-    def _get_pending_operations(self, jobs, operation_names=None):
+    def _get_pending_groups(self, jobs, groups, exec_mode=False):
+        if exec_mode:
+            mode = 'exec'
+        else:
+            mode = 'run'
+        for job in jobs:
+            for group in groups:
+                if group.eligible(job):
+                    yield group.create_job_operation(job, mode=mode, index=0)
+
+    def _get_pending_operations(self, jobs, operation_names=None,
+                                exec_mode=False):
         "Get all pending operations for the given selection."
         assert not isinstance(operation_names, six.string_types)
-        for op in self.next_operations(* jobs):
+        for op in self.next_operations(* jobs, exec_mode=exec_mode):
```

Comment:
Same as above, I think we should switch to `mode` as soon as possible, right after the CLI parsing level.